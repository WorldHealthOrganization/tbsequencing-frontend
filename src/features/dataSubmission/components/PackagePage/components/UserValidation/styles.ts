import { css } from '@emotion/react';

export const section = css({
  marginBottom: 24,
});

export const paper = css({
  marginBottom: 16,
});
