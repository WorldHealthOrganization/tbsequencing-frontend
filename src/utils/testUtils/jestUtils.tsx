import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
