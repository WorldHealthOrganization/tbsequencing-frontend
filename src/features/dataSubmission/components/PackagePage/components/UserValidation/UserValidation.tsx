/** @jsxImportSource @emotion/react */
import React from 'react';
import { toast } from 'react-toastify';
import * as styles from './styles';
import AppPaper from '../../../../../../components/AppPaper';
import H2 from '../../../../../../components/typography/H2';
import PrimaryText from '../../../../../../components/typography/PrimaryText';
import { AppButton } from '../../../../../../components/AppButton/AppButton';
import ActivityIndicator from '../../../../../../components/ActivityIndicator';
import { useVerifyUserMutation } from '../../../../../../services/indentityApi/identityApi';

interface Props {
  validationInProgress: boolean;
  isLoading: boolean;
}

export const UserValidation = ({ validationInProgress, isLoading }: Props) => {
  const validationMsg = validationInProgress
    ? 'You are waiting for verification. If it takes too long, contact us'
    : 'Before you start, you need to be validated by our team, '
    + 'which may take a few days. Provided the information is accurate, '
    + 'use the “Validate” button. If not, first change the data.';

  const [verifyUser] = useVerifyUserMutation();

  const handleValidateMeClick = async () => {
    try {
      await verifyUser().unwrap();
    } catch (error) {
      toast.error('There was an error trying to validate user');
    }
  };

  if (isLoading) {
    return <ActivityIndicator centered />;
  }

  return (
    <AppPaper style={styles.paper}>
      <H2 style={styles.section}>User validation</H2>
      <PrimaryText style={styles.section}>{validationMsg}</PrimaryText>
      <AppButton
        disabled={validationInProgress}
        onClick={handleValidateMeClick}
        variant="contained"
        size="large"
      >
        Validate me
      </AppButton>
    </AppPaper>
  );
};

export default UserValidation;
