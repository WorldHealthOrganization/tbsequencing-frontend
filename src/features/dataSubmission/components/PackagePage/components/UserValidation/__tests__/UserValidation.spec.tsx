import React from 'react';
import { getTreeJSONWithRedux, render, screen } from '../../../../../../../utils/testUtils/jestUtils';
import UserValidation from '../UserValidation';

describe('UserValidation component test', () => {
  test('renders UserValidation component without error', () => {
    render(
      <UserValidation validationInProgress />,
    );
  });

  it('Should render validation message in case of validation', () => {
    render(
      <UserValidation validationInProgress />,
    );

    screen.getByText('You are waiting for verification. If it takes too long, contact us');
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <UserValidation validationInProgress />,
    );

    expect(tree).toMatchSnapshot();
  });
});
