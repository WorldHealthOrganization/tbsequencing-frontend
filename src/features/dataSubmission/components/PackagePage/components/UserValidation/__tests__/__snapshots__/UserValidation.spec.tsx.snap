// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserValidation component test Should render correctly 1`] = `
<div
  className="css-fmmjw3-AppPaper"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-f2rd29-MuiTypography-root-BaseText"
  >
    User validation
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1dmeqf6-MuiTypography-root-BaseText"
  >
    You are waiting for verification. If it takes too long, contact us
  </p>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled css-l4q5pu-MuiButtonBase-root-MuiButton-root-AppButton"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
    >
      Validate me
    </p>
  </button>
</div>
`;
